core.js

```diff
@@ -0,0 +1,131 @@
+/*$AMPERSAND_VERSION*/
+var result = require('lodash.result');
+var defaults = require('lodash.defaults');
+var includes = require('lodash.includes');
+var assign = require('lodash.assign');
+var qs = require('qs');
+
+module.exports = function (xhr) {
+
+  // Throw an error when a URL is needed, and none is supplied.
+  var urlError = function () {
+      throw new Error('A "url" property or function must be specified');
+  };
+
+  // Map from CRUD to HTTP for our default `Backbone.sync` implementation.
+  var methodMap = {
+      'create': 'POST',
+      'update': 'PUT',
+      'patch':  'PATCH',
+      'delete': 'DELETE',
+      'read':   'GET'
+  };
+
+  return function (method, model, options) {
+      var type = methodMap[method];
+      var headers = {};
+
+      // Default options, unless specified.
+      defaults(options || (options = {}), {
+          emulateHTTP: false,
+          emulateJSON: false,
+          // overrideable primarily to enable testing
+          xhrImplementation: xhr
```

Comment:
I looked for documentation mentioning this option, but it seems undocumented. For the sake of being more cross-platform we might consider changing it to one of these:
requestOverride
requestEngine
httpRequestImplementation